firmware/library/L2_HAL/temperature.hpp

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdio>
+#include "L1_Drivers/i2c.hpp"
+
+class TemperatureInterface
+{
+ public:
+  virtual bool Initialize()                       = 0;
+  virtual int GetCelsius()                        = 0;
```

Comment:
What size int are you using for this? Please use a specific integer size for the values here. Please see <cstdint> for more details.